test/suites/beta/scale/deprovisioning_test.go

```diff
@@ -0,0 +1,788 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scale_test
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"knative.dev/pkg/ptr"
+
+	corev1beta1 "github.com/aws/karpenter-core/pkg/apis/v1beta1"
+	"github.com/aws/karpenter-core/pkg/test"
+	"github.com/aws/karpenter/pkg/apis/v1beta1"
+	"github.com/aws/karpenter/pkg/controllers/interruption/messages"
+	"github.com/aws/karpenter/pkg/controllers/interruption/messages/scheduledchange"
+	awstest "github.com/aws/karpenter/pkg/test"
+	"github.com/aws/karpenter/pkg/utils"
+	"github.com/aws/karpenter/test/pkg/debug"
+	"github.com/aws/karpenter/test/pkg/environment/aws"
+)
+
+const (
+	deprovisioningTypeKey = "testing/deprovisioning-type"
+	consolidationValue    = "consolidation"
+	emptinessValue        = "emptiness"
+	expirationValue       = "expiration"
+	noExpirationValue     = "noExpiration"
+	driftValue            = "drift"
+	noDriftValue          = "noDrift"
+)
+
+const (
+	multipleDeprovisionersTestGroup = "multipleDeprovisioners"
+	consolidationTestGroup          = "consolidation"
+	emptinessTestGroup              = "emptiness"
+	expirationTestGroup             = "expiration"
+	driftTestGroup                  = "drift"
+	interruptionTestGroup           = "interruption"
+
+	defaultTestName = "default"
+)
+
+// disableProvisioningLimits represents limits that can be applied to a nodePool if you want a nodePool
+// that can deprovision nodes but cannot provision nodes
+var disableProvisioningLimits = corev1beta1.Limits{
+	v1.ResourceCPU:    resource.MustParse("0"),
+	v1.ResourceMemory: resource.MustParse("0Gi"),
+}
+
+var _ = Describe("Deprovisioning", Label(debug.NoWatch), Label(debug.NoEvents), func() {
+	var nodePool *corev1beta1.NodePool
+	var nodeClass *v1beta1.EC2NodeClass
+	var deployment *appsv1.Deployment
+	var deploymentOptions test.DeploymentOptions
+	var selector labels.Selector
+	var dsCount int
+
+	BeforeEach(func() {
+		env.ExpectSettingsOverriddenLegacy(map[string]string{"featureGates.driftEnabled": "true"})
```

Comment:
```suggestion
		env.ExpectSettingsOverridden(v1.EnvVar{Name: "FEATURE_GATES", Value: "Drift=True"})
```

Probably want to use the updated way of upserting the options